pkg/controller/workload/job/job_webhook.go

```diff
@@ -69,6 +71,13 @@ func (w *JobWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log := ctrl.LoggerFrom(ctx).WithName("job-webhook")
 	log.V(5).Info("Applying defaults", "job", klog.KObj(job))
 
+	if owner := metav1.GetControllerOf(job); owner != nil && workloadcommon.IsMPIJob(owner) {
+		if job.Annotations == nil {
+			job.Annotations = make(map[string]string)
+		}
+		job.Annotations[constants.ParentWorkloadAnnotation] = workloadcommon.GetWorkloadNameForMPIJob(owner.Name)
```

Comment:
Yes, I'm hesitant too, but I think it would be convenient to put as much support as possible into the already existing webhook for Job. I have followed the suggestion on `GetWorkloadNameForOwner` for now.